### PR TITLE
Fixes #28422 - use jest instead of jest-cli

### PR DESCRIPTION
**Pull request by amirfefer**
in foreman core we use `jest-cli` for testing, but in plugins, we use regular jest
due to moving it to `@foreman/env` we should keep it unified

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "lint": "./node_modules/.bin/eslint -c .eslintrc webpack/ script/",
     "foreman-js:link": "./script/npm_link_foreman_js.sh",
     "postlint": "./script/npm_lint_plugins.js",
-    "test": "node --max_old_space_size=8192 node_modules/.bin/jest --no-cache --config config/jest.config.js",
+    "test": "jest",
```
Comment (ekohl) [inline package.json]: Can't we specify `jest --config config/jest.config.js` to keep our root directory a bit cleaner?

**Comment by amirfefer** [comment]
Removing the `no-cache` flag reduced the running time significantly
```
Test Suites: 4 failed, 201 passed, 205 total
Tests:       6 failed, 3 skipped, 819 passed, 828 total
Snapshots:   2 obsolete, 539 passed, 539 total
Time:        81.939s, estimated 190s
```
VS
```
Test Suites: 4 failed, 201 passed, 205 total
Tests:       6 failed, 3 skipped, 819 passed, 828 total
Snapshots:   2 obsolete, 539 passed, 539 total
Time:        181.395s
```

According to jest docs:
> Whether to use the cache. Defaults to true. Disable the cache using --no-cache. Note: the cache should only be disabled if you are experiencing caching related problems. On average, disabling the cache makes Jest **at least two times slower.**

**Comment by amirfefer** [comment]
[foreman test]

**Review by mmoll — DISMISSED** [review]

**Comment by Ron-Lavi** [comment]
@amirfefer can you rebase?

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "lint": "./node_modules/.bin/eslint -c .eslintrc webpack/ script/",
     "foreman-js:link": "./script/npm_link_foreman_js.sh",
     "postlint": "./script/npm_lint_plugins.js",
-    "test": "node --max_old_space_size=8192 node_modules/.bin/jest --no-cache --config config/jest.config.js",
+    "test": "jest --config config/jest.config.js",
```
Comment (Ron-Lavi) [inline package.json:11]: This is absolutely amazing !
with cache:
tests on first run: 160 seconds,
on second time: 80 seconds,
on third time: 40 seconds,
on 4th time: 18 seconds

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, thanks @amirfefer!

**Pull request merged**